contributingGuides/HYBRID_APP.md

```diff
@@ -0,0 +1,212 @@
+There are multiple thing that may cause the build to fail. This is a simple checklist that may help you with debugging:
+1. Read the error carefully - sometimes it explains the problem very well
+2. Look for a similar error in the Troubleshooting section
+3. Pull the newest `main`
+4. Check if you have a recent version of the `Mobile-Expensify` submodule
+	- if not, execute `git submodule update`, and rerun the build
+5. Clean the cache by executing `npm run clean` (with proper platform flags)
+If the problem still exists, please share the error on the open source channel
+### What if I want to build a standalone NewDot?
+There are additional helper scripts, that will help you to run a standalone NewDot, even if you have access to `Mobile-Expensify`, and the default behaviour of the scripts builds HybridApp. 
+
+In this case you can do the following:
+1. `rm -rf node_modules`
+2. `npm i-standalone`
+3. `npm run pod-install-standalone` (required only if you build iOS)
+4. `npm run ios-standalone` or `npm run android-stanalone`
+
+Alternatively, you may notice in `package.json` that all these scripts are based on `STANDALONE_NEW_DOT` environment variable. If you feel confident you can do `export STANDALONE_NEW_DOT=true` to change the default behaviour in the current process of the terminal, and use regular commands (`npm i`, `npm run android`, `npm run ios` etc.) to build NewDot. 
+
+### Do I need to clean cache and rebuild the app?
+It's a valid question, especially because clean builds may take some time. On the way I've noticed that many developers tend to rebuild the app from scratch, even though in some cases it is unnecessary. In this case, when should you rebuild the app?
+1. Whenever any code located in `./Mobile-Expensify` has changed - this means that we need to recompile OldDot/native code
+2. Whenever you've pulled the newest main - not always necessary, but usually we don't analise what code has just been merged by `git pull`
+3. Whenever `package-lock.json` has changed - this may indicate that some packages with native code were bumped (however it's not always necessary, see [[#Should I rebuild HybridApp after bumping a `node_module`?]])
+4. Whenever you've updated `.env` files
+This means that if you changed only React Native code, and didn't pull any changes, the rebuilt is probably not necessary. If something doesn't work, you can always restart the Metro bundler using the following command `npm run start --reset-cache`
+### Should I rebuild HybridApp after bumping a `node_modules` library?
+The `package-lock.json` file contains information about exact versions of `node_modules` that will be installed on your machine. If you've bumped a dependency on your PR you can easily see if you would need to rebuild the app. In this case go to the `./node_modules/<PACKAGE_NAME>`, and see if there are any Objective-C, Swift, Java, Kotlin or C++ files. Usually they are located in `ios` or `android` folders.
```

Comment:
> If you've bumped a dependency on your PR you can easily see if you would need to rebuild the app. In this case go to the `./node_modules/<PACKAGE_NAME>`, and see if there are any Objective-C, Swift, Java, Kotlin or C++ files.

If you've bumped a dependency on your PR you can easily check to see if you would need to rebuild the app by going to the `./node_modules/<PACKAGE_NAME>`, and seeing if there are any Objective-C, Swift, Java, Kotlin or C++ files.